### PR TITLE
Design custom posts

### LINKED ISSUE #24: Design Custom Post (Single post) pages.
Once custom posts for anime are created (#2), site should be populated with 3-10 different anime posts

### PR DESCRIPTION
**Pull request by jefroy**
this branch deals with custom post pages. I aim to display the data and related fields appropriately.

### CODE REVIEW

File: public/wp-content/themes/newsup-child/single-anime.php [hunk 1]
```diff
@@ -7,6 +7,7 @@
 <!-- =========================
      Page Content Section
 ============================== -->
+single-anime.php
```
Comment (shottah) [inline public/wp-content/themes/newsup-child/single-anime.php]: Why is this a commit. This is a bug.

File: public/wp-content/themes/newsup/single.php [hunk 2]
```diff
@@ -7,6 +7,7 @@
 <!-- =========================
      Page Content Section      
 ============================== -->
+single.php
```
Comment (shottah) [inline public/wp-content/themes/newsup/single.php]: This also. Why is this here? This line

**Review by shottah — DISMISSED** [review]
This commit gave me terminal cancer.

File: public/wp-content/themes/newsup-child/single-anime.php (hunk 1 above)
Comment (jefroy) [reply public/wp-content/themes/newsup-child/single-anime.php]: i was testing something, ill remove it before i request to merge

File: public/wp-content/themes/newsup/single.php (hunk 2 above)
Comment (jefroy) [reply public/wp-content/themes/newsup/single.php]: plz :(

**Review by shottah — COMMENTED** [review]
Appreciate the work, well done it's all there, but one question, why didn't you use the style block that already existed.

The one that has the "date" and "adminuser" as post author can be reappropriated for "Date Written" and "Anime Author Name".

Also what I would do is use the styles that wrap the "Uncategorized" icons around the site to encapsulate the genres as a list under the Anime title.

**Comment by jefroy** [comment]
> Appreciate the work, well done it's all there, but one question, why didn't you use the style block that already existed.
> 
> The one that has the "date" and "adminuser" as post author can be reappropriated for "Date Written" and "Anime Author Name".
> 
> Also what I would do is use the styles that wrap the "Uncategorized" icons around the site to encapsulate the genres as a list under the Anime title.

Im gonna style it next, i just wanted to make sure the content was working.

Ill try to modify the main card to your liking as well, thanks for pointing that out

**Comment by shottah** [comment]
Make sure you remove the `sql/local.sql` file from the push as it is no longer part of the repo. Also, the `newsup/single.php` diff is caused by an update to newsup theme, so ensure to use the updated version.

**Comment by jefroy** [comment]
> Make sure you remove the `sql/local.sql` file from the push as it is no longer part of the repo. Also, the `newsup/single.php` diff is caused by an update to newsup theme, so ensure to use the updated version.

noted

**Comment by jefroy** [comment]
> Appreciate the work, well done it's all there, but one question, why didn't you use the style block that already existed.
> 
> The one that has the "date" and "adminuser" as post author can be reappropriated for "Date Written" and "Anime Author Name".
> 
> Also what I would do is use the styles that wrap the "Uncategorized" icons around the site to encapsulate the genres as a list under the Anime title.

please refer to the image below @matabeitt 

image

1: should this be removed? maybe we could use javascript to freeze it, so that when the user scrolls down, it remains snapped to the top of the screen?

2: does this seem more appropriate?

**Comment by jefroy** [comment]
image

@matabeitt im trying to relate anime reviews to anime posts, but i think this needs to be a relationship type. Are you certain this is correct?

**Comment by shottah** [comment]
This is from the Review side. So a review can only be about one Anime, not multiple Anime. Try looping over all Reviews where their review_anime = target 

Or you can look up the return type if Post Object from ACF. 

Remove the Title Bar on top and leave the one near the post author.

**Comment by jefroy** [comment]
> This is from the Review side. So a review can only be about one Anime, not multiple Anime. Try looping over all Reviews where their review_anime = target
> 
> Or you can look up the return type if Post Object from ACF.
> 
> Remove the Title Bar on top and leave the one near the post author.

gotcha, thanks

File: public/wp-content/themes/newsup-child/single-anime.php [hunk 3]
```diff
@@ -123,6 +128,71 @@
...
+                          )
+                      )
+                  ));
+                  $reviews_posts = array();
+                  foreach ($reviews_array as $item) {
+                      array_push($reviews_posts, setup_postdata($item));
+                  }
+                  echo (gettype($reviews_query)) . '<br>'; //array of associated review posts
+                  echo (gettype($reviews_array)) . '<br>';
+                  echo (gettype($reviews_posts)) . '<br>';
```
Comment (shottah) [inline public/wp-content/themes/newsup-child/single-anime.php]: This needs to be removed. This echos to the front end.

File: public/wp-content/themes/newsup/single.php [hunk 4]
```diff
@@ -1,3 +1,4 @@
+<<<<<<< HEAD
```
Comment (shottah) [inline public/wp-content/themes/newsup/single.php]: This merge conflict was not resolved properly. You need to pick whether to keep code from "HEAD" or from "Master" and now, both HEAD and Master code is in this file.

**Review by shottah — CHANGES_REQUESTED** [review]
This cannot merge into master.

File: public/wp-content/themes/newsup-child/single-anime.php (hunk 3 above)
Comment (jefroy) [reply public/wp-content/themes/newsup-child/single-anime.php]: yea mb, that was for testing. i will remove it for sure

File: public/wp-content/themes/newsup/single.php (hunk 4 above)
Comment (jefroy) [reply public/wp-content/themes/newsup/single.php]: oh.. ill replace it with the file from master now.

File: public/wp-content/themes/newsup/index-banner.php [hunk 5]
```diff
@@ -5,23 +5,23 @@
...
+<!--          </div>-->
+<!--        </div>-->
+<!--      </div>-->
+<!--    </div>-->
+<!--  --><?php //$newsup_remove_header_image_overlay = get_theme_mods('remove_header_image_overlay',true);
+//if($newsup_remove_header_image_overlay == true){ ?>
+<!--  </div>-->
+<?php //} ?>
+<!--</div>-->
+<!--<div class="clearfix"></div>-->
```
Comment (shottah) [inline public/wp-content/themes/newsup/index-banner.php]: It would be best if critical changes were only made to the `newsup-child` as to push changes to the website requires us to FTP files across. It would be a pain in the ass doing it for multiple themes. See if you can achieve this using only the child theme. (hint: Copy down the files into child from parent, and remove the lines of code there)

**Review by shottah — DISMISSED** [review]

File: public/wp-content/themes/newsup/index-banner.php (hunk 5 above)
Comment (jefroy) [reply public/wp-content/themes/newsup/index-banner.php]: oh right, my bad. gonna fix this now. thanks for reminding me. im not used to the child directory xD

Comment (shottah) [reply public/wp-content/themes/newsup/index-banner.php]: I doing this for Phaedra tbh. A production theme is a LOT to find differences in.

Comment (jefroy) [reply public/wp-content/themes/newsup/index-banner.php]: yea, navigating all the built in functions and widgets kinda spooky too xD

but it slight manageable using the inspector tool on chrome and searching the entire project :( 

making do with it oui

Comment (shottah) [reply public/wp-content/themes/newsup/index-banner.php]: Have you seen the site using 1 alone and 2 alone? (Referring to how you numbered the title sections. Which one looks better to you? Just use whatever you think is best. I guessed 2 would look better but you should make that call.

Comment (jefroy) [reply public/wp-content/themes/newsup/index-banner.php]: it looks much cleaner with just the one title IMO

image

im trying to reduce clutter, but not omit anything that we can potentially use. That extra title bar rel outtaz imo, but if you or asa thinks it belongs there, ill put it back.

**Comment by jefroy** [comment]
hello good day, i think im done
@matabeitt  @asayee0 

if yall wanna give it a quick look

reddy2merge

**Comment by shottah** [comment]
1. Use sidebar to display review links.
2. Use "You Missed" to display anime related to current by genre (any order).

**Comment by jefroy** [comment]
> 1. Use sidebar to display review links.
> 2. Use "You Missed" to display anime related to current by genre (any order).

ok will try

File: public/wp-content/themes/.gitignore [hunk 6]
```diff
@@ -0,0 +1,4 @@
+# ignore all themes except newsup-based themes and .gitignore
+*
+!.gitignore
+!public/wp-content/themes/newsup*
```
Comment (asayee0) [inline public/wp-content/themes/.gitignore:4]: You need to add the all-in-one-wp-migration to the .gitignore because about 170+ of these changed files are from that.

File: public/wp-content/themes/newsup-child/single-anime.php [hunk 7]
```diff
@@ -66,18 +70,148 @@
...
-            <a class="mg-author-pic" href="<?php echo esc_url(get_author_posts_url( get_the_author_meta( 'ID' ) ));?>"><?php echo get_avatar( get_the_author_meta( 'ID') , 150); ?></a>
-                <div class="media-body">
-                  <h4 class="media-heading"><?php esc_html_e('By','newsup'); ?> <a href "<?php echo esc_url(get_author_posts_url( get_the_author_meta( 'ID' ) ));?>"><?php the_author(); ?></a></h4>
-                  <p><?php the_author_meta( 'description' ); ?></p>
-                </div>
-              <?php } ?>
-            </div><?php $newsup_enable_related_post = esc_attr(get_theme_mod('newsup_enable_related_post','true'));
+<!--                    list genres and anime information here-->
+                    <?php
+                    $genres = get_field('anime_genres'); // relationship (other post type)
```
Comment (asayee0) [inline public/wp-content/themes/newsup-child/single-anime.php:70]: Add a condition here to *not* load a relationship if there are no genres defined. I don't have any defined on my db and I'm getting an error.

error

File: public/wp-content/themes/newsup-child/single-genre.php [hunk 8]
```diff
@@ -7,169 +7,229 @@
 <!-- =========================
      Page Content Section
 ============================== -->
+<!--<p>single-anime.php -> in child dir</p>-->
```
Comment (asayee0) [inline public/wp-content/themes/newsup-child/single-genre.php]: Remember to remove all your test code

**Review by asayee0 — CHANGES_REQUESTED** [review]
== In Summary ==
- [x] Add WP Migration plugin to gitignore
- [x] Change single-anime to not relate to genre if it doesn't exist
- [x] Remove your test code

File: public/wp-content/themes/newsup-child/single-anime.php (hunk 7 above)
Comment (shottah) [reply public/wp-content/themes/newsup-child/single-anime.php:70]: anime_genre is a required field, but sure, add this.

File: public/wp-content/themes/newsup-child/single-genre.php (hunk 8 above)
Comment (shottah) [reply public/wp-content/themes/newsup-child/single-genre.php]: I will be going thru a full documentation overhaul after these round of branches are done.

File: public/wp-content/themes/newsup-child/single-anime.php (hunk 7 above)
Comment (asayee0) [reply public/wp-content/themes/newsup-child/single-anime.php:70]: Yeah i think it's still worth it to put since I have the ability to make an anime without the required field.
We could probably make a new issue for that.

Comment (jefroy) [reply public/wp-content/themes/newsup-child/single-anime.php:70]: that error not making any sense to me

image

i have checks for all the query objects and all the grabbed fields, 

i think your error is something else, maybe a db migration issue.
do you want me to use a try catch when i call get_field() ?

i also dont know how to make an anime post without the required fields in order to test the issue XD

Comment (shottah) [reply public/wp-content/themes/newsup-child/single-anime.php:70]: `Call to undefined function get_field()` is saying that you do not have ACF installed. ACF plugin defines `get_field()`.

Comment (jefroy) [reply public/wp-content/themes/newsup-child/single-anime.php:70]: > `Call to undefined function get_field()` is saying that you do not have ACF installed. ACF plugin defines `get_field()`.

yea, i was thinking this too

File: public/wp-content/themes/.gitignore (hunk 6 above)
Comment (jefroy) [reply public/wp-content/themes/.gitignore:4]: i added the entire plugins folder to my gitignore

does that solve the problem?

File: public/wp-content/themes/newsup-child/single-genre.php (hunk 8 above)
Comment (jefroy) [reply public/wp-content/themes/newsup-child/single-genre.php]: done

File: public/wp-content/themes/.gitignore (hunk 6 above)
Comment (jefroy) [reply public/wp-content/themes/.gitignore:4]: > 1. Use sidebar to display review links.
> 2. Use "You Missed" to display anime related to current by genre (any order).

1. i have no clue how to modify the widgets to take in custom types
2. i loaded in anime instead of posts, but i dont know how to pull the genre of the page since its in the footer. im thinking this is very non-essential, and if you want a related anime section, i could do that in another card within single-anime.php. leme know what you think :)

File: public/wp-content/themes/newsup/readme.txt [hunk 9]
```diff
@@ -159,4 +159,11 @@ License: CC0 Public Domain
 * Fixed sanitize issue.
 
 = Version 2.6
-* Added Edit link and fix archive page title issue.
\ No newline at end of file
+<<<<<<< HEAD
```
Comment (asayee0) [inline public/wp-content/themes/newsup/readme.txt:162]: Merge errors to be removed.

This file isn't really relevant so it's not necessary, but still good for future note.

File: public/wp-content/themes/newsup-child/single-anime.php (hunk 7 above)
Comment (asayee0) [reply public/wp-content/themes/newsup-child/single-anime.php:70]: Yep yall right. I had it installed, but I didn't have the relationship set up. All good.

File: public/wp-content/themes/.gitignore (hunk 6 above)
Comment (asayee0) [reply public/wp-content/themes/.gitignore:4]: > i added the entire plugins folder to my gitignore
> 
> does that solve the problem?

Yup

> > 1. Use sidebar to display review links.
> > 2. Use "You Missed" to display anime related to current by genre (any order).
> 
> 1. i have no clue how to modify the widgets to take in custom types
> 2. i loaded in anime instead of posts, but i dont know how to pull the genre of the page since its in the footer. im thinking this is very non-essential, and if you want a related anime section, i could do that in another card within single-anime.php. leme know what you think :)

Yeah I'm thinking we could just skip on this part unless anyone wants to tinker around with it just for the experience.

File: public/wp-content/themes/newsup/readme.txt (hunk 9 above)
Comment (jefroy) [reply public/wp-content/themes/newsup/readme.txt:162]: oh

File: public/wp-content/themes/newsup-child/single-anime.php [hunk 10]
```diff
@@ -138,9 +138,10 @@
 <!--                    print associated anime reviews in another card-->
                   <?php
                   $query_all = array(
-                      'posts_per_page' => -1,
+                      'posts_per_page' => 15,
```
Comment (shottah) [inline public/wp-content/themes/newsup-child/single-anime.php:141]: Is it 15 or is it 10?

File: .gitignore [hunk 11]
```diff
@@ -31,6 +31,8 @@ node_modules/
 
 # ???? consider tracking the entire plugins directory
 public/wp-content/cache/*
+public/wp-content/plugins/*
+
 public/wp-content/plugins/sg-cachepress/*
 public/wp-content/plugins/wordpress-starter/*
```
Comment (shottah) [inline .gitignore]: This can be removed.

File: public/wp-content/themes/newsup-child/single-anime.php [hunk 12]
```diff
@@ -88,10 +88,10 @@
                         </div>
                         <div class="media-body">
                             <h2>Author</h2>
-                            <p><b><?php echo $author; ?></b></p>
+                            <p><b><?php if($author) echo $author; ?></b></p>
                             <hr>
                             <h2>Release Date</h2>
-                            <p><b><?php echo $release_date; ?></b></p>
+                            <p><b><?php if($release_date) echo $release_date; ?></b></p>
```
Comment (shottah) [inline public/wp-content/themes/newsup-child/single-anime.php]: If it have no author or date you going to just leave it blank?

File: public/wp-content/themes/newsup-child/single-anime.php (hunk 10 above)
Comment (jefroy) [reply public/wp-content/themes/newsup-child/single-anime.php:141]: when i use 10, it brings up a random range under 10, so i decided 15 would be good. even tho its on 15, it brings up 9-11 or so

File: public/wp-content/themes/.gitignore (hunk 6 above)
Comment (jefroy) [reply public/wp-content/themes/.gitignore:4]: agreed.

File: public/wp-content/themes/newsup-child/single-anime.php (hunk 12 above)
Comment (jefroy) [reply public/wp-content/themes/newsup-child/single-anime.php]: good eye, ill refactor this xD tyyy

File: public/wp-content/themes/newsup-child/footer.php [hunk 13]
```diff
@@ -21,7 +22,14 @@
                 </div>
             </div>
             <?php }
-            $newsup_you_missed_loop = new WP_Query(array( 'post_type' => 'post', 'posts_per_page' => 4, 'order' => 'DESC',  'ignore_sticky_posts' => true));
+            $you_missed_args = array(
+	            'post_type' => 'anime',
+                'posts_per_page' => 4,
+                'order' => 'DESC',
+                'orderby' => 'rand',
```
Comment (shottah) [inline public/wp-content/themes/newsup-child/footer.php:29]: Put this is order of newest to oldest, make it how many ever fit on a full screen medium viewport,

Comment (shottah) [reply public/wp-content/themes/newsup-child/footer.php:29]: and change it to "You May have missed" instead of "You missed"

**Review by shottah — CHANGES_REQUESTED** [review]
Okay we are mostly looking in order these are some changes that will be made when we make the call anyways.

File: .gitignore (hunk 11 above)
Comment (jefroy) [reply .gitignore]: image

is this what you wanted?

File: public/wp-content/themes/newsup-child/single-anime.php (hunk 12 above)
Comment (jefroy) [reply public/wp-content/themes/newsup-child/single-anime.php]: gateeeem

File: .gitignore (hunk 11 above)
Comment (shottah) [reply .gitignore]: If you're ignoring all plugins, you don't need to block extra plugins by name. Please fix.
